### ISSUE TITLE
fileInput drag and drop failed with rHandsontableOutputin the UI

### ISSUE THREAD

**Issue by dylancis** [rstudio/shiny#1962]
It took me a while to pinpoint where the issue is coming from but I have now a reproducible example.
shiny fileInput component allows drag and drop file functionality. However this option is cancelling out while the ui contains a rHandsontableOutput widget.

Please check app.R
`
library(shiny)
library(rhandsontable)

ui <- fluidPage(

titlePanel("Uploading Files"),

sidebarLayout(

sidebarPanel(
  
  # Input: Select a file ---- this is where the issue is
  fileInput("file1", "Choose CSV File",
            multiple = TRUE,
            accept = c("text/csv",
                       "text/comma-separated-values,text/plain",
                       ".csv")),
  
  tags$hr(),
  
  checkboxInput("header", "Header", TRUE),
  
  radioButtons("sep", "Separator",
               choices = c(Comma = ",",
                           Semicolon = ";",
                           Tab = "\t"),
               selected = ","),
  
  radioButtons("quote", "Quote",
               choices = c(None = "",
                           "Double Quote" = '"',
                           "Single Quote" = "'"),
               selected = '"'),
  
  tags$hr(),
  radioButtons("disp", "Display",
               choices = c(Head = "head",
                           All = "all"),
               selected = "head")
  
),

mainPanel(
  
  tableOutput("contents"),
 # try to drag and drop any file in fileInput field with and without the line below
  rHandsontableOutput("hot")
)
)
)

server <- function(input, output) {

output$contents <- renderTable({
req(input$file1)

df <- read.csv(input$file1$datapath,
               header = input$header,
               sep = input$sep,
               quote = input$quote)

if(input$disp == "head") {
  return(head(df))
}
else {
  return(df)
}
})

}

shinyApp(ui, server)`

Looking at the developer tools from Chrome I see that every time I tried to drag a file into Shiny app get triggered:
![image](https://user-images.githubusercontent.com/4976433/36787626-b1a8e76c-1c82-11e8-830f-4d912e649951.png)

**Comment by dylancis** [comment]
This is an annoying issue, an idea on how to fix it please?

**Comment by bborgesr** [comment]
MRE

```r
library(shiny)

ui <- fluidPage(
  fileInput("file1", "Choose CSV File", multiple = TRUE,
    accept = c("text/csv", "text/comma-separated-values,text/plain", ".csv"))
  tableOutput("contents"),
  rhandsontable::rHandsontableOutput("hot")
)

server <- function(input, output, session) {
  output$contents <- renderTable({
    req(input$file1)
    read.csv(input$file1$datapath)
  })
}

shinyApp(ui, server)
```

The problem seems to happen due to the `collection.size()` calls here: https://github.com/rstudio/shiny/blob/master/srcjs/input_binding_fileinput.js#L326-L341

**Comment by jcheng5** [comment]
Rhandsontable includes jQuery 3.x, maybe that trumps our copy and doesn't have exactly the same API?

**Comment by jcheng5** [comment]
https://api.jquery.com/size/

**Comment by alandipert** [comment]
@dylancis I made a branch to explore @jcheng5's theory. Please try it and let us know if it works:

`devtools::install_github("rstudio/shiny", ref = "alan/bugfix/dndfix")`

**Comment by dylancis** [comment]
Hi @alandipert ,
It works perfectly. I also checked the handson tables see if there was no break there and all fine.
I have also tried multiple files drag and also good there. Hope it will make its way to master branch soon.
Many thanks, Dylan

**Comment by bborgesr** [comment]
Closed by #2005
